mh/pat.py

```diff
@@ -83,6 +73,18 @@ def circle_broadcast(self, circle, packet_id, data, seq,
         for handler in handlers:
             handler.try_send_packet(packet_id, data, seq)
 
+    def layer_broadcast(self, session, packet_id, data, seq,
+                        exclude_self=True):
+        # type: (Session, int, bytes, int, bool) -> None
+        self.broadcast(session.get_layer_players(), packet_id, data, seq, 
+                       None if exclude_self else session)
+
+    def circle_broadcast(self, circle, packet_id, data, seq,
+                         session=None):
+        # type: (db.Circle, int, bytes, int, Session|None) -> None
+        self.broadcast(circle.players, packet_id, data, seq, session)
```

Comment:
I don't know if that matters but `with circle.lock():` was removed when the `broadcast` helper was introduced.